scripts/classification/imagenet/train_imagenet.py

```diff
@@ -105,6 +106,10 @@ def parse_args():
                         help='name of training log file')
     parser.add_argument('--use-gn', action='store_true',
                         help='whether to use group norm.')
+    parser.add_argument('--amp', action='store_true',
+                    help='Use MXNet AMP for mixed precision training.')
+    parser.add_argument('--auto-layout', action='store_true',
```

Comment:
Could you also add an option like `--target-dtype` since now we not only have `float16` for amp, but `bfloat16`. Then, we can pass `target-dtype` to `amp.init()` to enable float16/bfloat16 training for GPU and CPU respectively. Thanks.